server/src/main.rs

```diff
@@ -1,10 +1,58 @@
 use warp::Filter;
+use serde::{Serialize, Deserialize};
 
 #[tokio::main]
 async fn main() {
-    let checkin = warp::path!("v1" / "checkin").map(|| "checkin");
-    let app = checkin;
+    let api = filters::checkins();
+let routes = api.with(warp::log("checkins"));
 
-    warp::serve(app)
+    warp::serve(routes)
         .run(([127, 0, 0, 1], 3000)).await;
 }
+
```

Comment:
Seems alright. You'll have to look at the functions that I'm working on too though.